source/slang/slang-parser.cpp

```diff
@@ -4972,6 +4972,24 @@ static void CompleteDecl(
                     staticConstDecl->nameAndLoc = enumCase->nameAndLoc;
                     addModifier(staticConstDecl, parser->astBuilder->create<HLSLStaticModifier>());
                     addModifier(staticConstDecl, parser->astBuilder->create<ConstModifier>());
+
+                    // Copy visibility modifiers from the enum declaration
+                    if (auto visibilityModifier = modifiers.findModifier<VisibilityModifier>())
+                    {
+                        if (as<PublicModifier>(visibilityModifier))
```

Comment:
you can call `astBuilder->createByNodeType(visibilityModifier.astNodeType)` instead of doing individual `if` for each type of modifier.